packages/astro/src/@types/astro.ts

```diff
@@ -2390,15 +2392,33 @@ interface AstroSharedContext<
 	currentLocale: string | undefined;
 }
 
+/**
+ * The `APIContext` is the object made available to endpoints and middleware.
+ * It is a subset of the `Astro` global object available in pages.
+ * 
+ * [Reference](https://docs.astro.build/en/reference/api-reference/#endpoint-context)
+ */
 export interface APIContext<
 	Props extends Record<string, any> = Record<string, any>,
 	APIParams extends Record<string, string | undefined> = Record<string, string | undefined>,
 > extends AstroSharedContext<Props, Params> {
+	/**
+	 * The site provided in the astro config, parsed as an instance of `URL`.
+	 * `undefined` if the site is not provided in the config.
```

Comment:
Worth mentioning if it contains `base` or not

Reply:
Good point.